components/studio/bin/hab-studio.sh

```diff
@@ -1076,6 +1076,35 @@ chown_artifacts() {
   fi
 }
 
+# **Internal**  Run a command which may fail without aborting whole script
+try() {
+  local status
+  if [[ $- == *e* ]]; then
+    # Disable `set -e` as briefly as possible to avoid masking other errors
+    set +e
+    $*
+    status=$?
+    set -e
+  else
+    $*
+    status=$?
+  fi
+
+  if [ $status -ne 0 ]; then
+    echo "Warning: '$*' failed with status $status"
+  fi
+}
+
+# **Internal** Unmount mount point if mounted
+# Don't abort script on umount failure
+# ARGS: [umount_options] <mount_point>
+try_umount() {
+  eval _mount_point=\$$# # getting the last arg is surprisingly hard
```

Comment:
Wow! That ought to work as far as I know. You could also use `awk` for this with

```sh
_mount_point="$(echo "$@" | awk '{print $NF}')"
```

Reply:
Yeah, it's surprising to me that there's not an easier way. I'll stick with this unless you think the `awk` version would be more easily understood by most.